### PR TITLE
casperjs: vendor phantomjs182

### PR DESCRIPTION
**Pull request by DomT4**
Closes #38632

### CODE REVIEW

**Comment by DomT4** [comment]
Failure *looks* unrelated. It's failing to find the stable version of Casper to test, which is kind of obvious.

```
Error: casperjs: failed
No such file or directory - /usr/local/Cellar/casperjs/1.0.4/bin/casperjs
```

Doing it outside of ` brew test ` is fine:

```
CasperJS version 1.1.0-beta3 at /usr/local/Cellar/casperjs/1.1-beta3/libexec, using phantomjs version 1.8.2
```

File: Library/Formula/casperjs.rb [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
-  head 'https://github.com/n1k0/casperjs.git'
+  head "https://github.com/n1k0/casperjs.git"
+
+  # For embedded Phantomjs
+  depends_on :macos => :snow_leopard
 
-  depends_on 'phantomjs'
+  # https://github.com/Homebrew/homebrew/pull/38632
+  resource "phantomjs" do
+    url "https://phantomjs.googlecode.com/files/phantomjs-1.8.2-macosx.zip"
```
Comment (n1k0) [inline Library/Formula/casperjs.rb]: Will this enforce using PhantomJS 1.8.2 only? Could we rather set 1.9.8 which is the latest 1.x version CasperJS actually supports, which many improvements over 1.8.2?

Comment (DomT4) [inline Library/Formula/casperjs.rb]: Yes. CasperJS disagrees with you on supporting 1.9.8. Install 1.9.8, run CasperJS, and:

` CasperJS 1.0.x does not support PhantomJS version >= 1.9 `

**Comment by mikemcquaid** [comment]
It's just the `devel` test that's failing. Would be good to have that fixed before we merge.

**Comment by DomT4** [comment]
Hilariously, it looks like the test breakage is down to case-sensitivity. The stable version outputs its PhantomJS version as ` PhantomJS version ` and the ` devel ` version seems to pump out the same thing as ` phantomjs version `. Sigh. Will cut down the expected output a little more to dance around that.

**Comment by cosmo0920** [comment]
Awesome!

File: Library/Formula/casperjs.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/casperjs.rb]: If you wanted to play around after installation though, you could just remove the ` export PATH ` line from ` bin/casperjs `.

**Comment by mikemcquaid** [comment]
Thanks @DomT4!

**Comment by indera** [comment]
Should we change the download link to https://github.com/ariya/phantomjs/archive/1.9.2.zip?

According to the official casperjs page (http://docs.casperjs.org/en/latest/installation.html) version 1.1.0-beta3 should work with phantomjs 1.9.2.

**Comment by DomT4** [comment]
#43180

**Comment by indera** [comment]
@DomT4   :+1:

**Pull request closed without merging**